sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/client/HttpClient.java

```diff
@@ -20,11 +23,46 @@ public interface HttpClient {
     Response<?> send(HttpRequest request);
 
     /**
-     * Creates a new {@link HttpClient} instance.
+     * Get an instance of the default {@link HttpClient} implementation.
      *
-     * @return A new {@link HttpClient} instance.
+     * <p>By default, the returned instance will be newly created and entirely unrelated to all other instances that
+     * were created previously. If the global variable {@code ENABLE_HTTP_CLIENT_SHARING} is configured to {@code true},
+     * a shared singleton {@link HttpClient} will be returned instead.</p>
+     *
+     * @return An instance of the default {@link HttpClient} implementation.
      */
-    static HttpClient createDefault() {
-        return HttpClientProviders.createInstance();
+    static HttpClient getDefault() {
+        if (Configuration.getGlobalConfiguration().get("ENABLE_HTTP_CLIENT_SHARING", Boolean.FALSE)) {
+            return DefaultHttpClientProvider.GlobalDefaultHttpClient.HTTP_CLIENT.getHttpClient();
+        }
```

Comment:
Should this check for `TRUE` to share the HTTP client?

Reply:
Also, we should flip the default to share the client and only use the env var to disable client sharing.

Reply:
Based on a separate discussion we had, we'll share `HttpClient` instances by default.